internal/internal_event_handlers.go

```diff
@@ -1311,7 +1356,20 @@ func (weh *workflowExecutionEventHandlerImpl) handleMarkerRecorded(
 				err = weh.dataConverter.FromPayloads(sideEffectDataPayload, &sideEffectDataID, &sideEffectDataContents)
 			}
 			if err == nil {
-				weh.mutableSideEffect[sideEffectDataID] = &sideEffectDataContents
+				counterHintPayload, ok := attributes.GetDetails()[mutableSideEffectCallCounterName]
+				var counterHint int
+				if ok {
+					err = weh.dataConverter.FromPayloads(counterHintPayload, &counterHint)
+				} else {
+					// An old version of the SDK did not write the counter hint so we have to assume.
```

Comment:
Maybe if we detect this scenario we should not write `counterHints` to maintain behaviour in a given workflow?

Reply:
I struggle to follow this code well enough to know, but so long as old behavior remains, any approach is fine by me